generated/nidaqmx/_library_interpreter.py

```diff
@@ -2285,7 +2319,8 @@ def read_analog_f64(
 
         error_code = cfunc(
             task, num_samps_per_chan, timeout, fill_mode,
-            ctypes.byref(read_array), ctypes.byref(samps_per_chan_read))
+            ctypes.byref(read_array), array_size_in_samps,
+            ctypes.byref(samps_per_chan_read))
         check_for_error(error_code)
         return read_array, samps_per_chan_read
```

Comment:
We need to handle numbers and strings differently from arrays.
- Python numbers are immutable, so the interpreter should always return them by value.
- Python strings are immutable, so the interpreter should always return them by value. However, in ctypes we use a mutable string buffer to interface with the DAQmx C API.
- Numpy arrays are mutable, and nidaqmx-python supports a number of functions that fill in preallocated arrays: https://github.com/ni/nidaqmx-python/blob/master/generated/nidaqmx/stream_readers.py . So we must pass them by reference, not by value.

This means:
- The interpreter signature for `read_analog_f64` should have a `read_array` parameter. It shouldn't return `read_array`.
- The interpreter signature for `read_analog_scalar_f64` should not have a `value` parameter. It should initialize `value = ctypes.c_double()`, pass `ctypes.byref(value)` to the C function, and return `value.value` like https://github.com/ni/nidaqmx-python/blob/master/generated/nidaqmx/_task_modules/channels/ai_channel.py#L5821
- The interpreter signature for `add_network_device` should not have a `device_name_out` parameter. It should initialize `device_name_out = ctypes.create_string_buffer(...)` and pass it to the C function like you are doing here, and it should return `device_name_out.value.encode('ascii')` like https://github.com/ni/nidaqmx-python/blob/master/generated/nidaqmx/_task_modules/channels/channel.py#L218

Reply:
BTW `read_array` isn't assigned before you call `ctypes.byref(read_array)`, so this probably returns `NameError: name 'read_array' is not defined`

Reply:
> BTW `read_array` isn't assigned before you call `ctypes.byref(read_array)`, so this probably returns `NameError: name 'read_array' is not defined`

I have raised another #287  on adding initialization logic for the output parameters used in the interpreter functions.

Reply:
> So is this ok to change the interpreter to accept the read array input

Yes.

The LibraryInterpreter should pass the `read_array` parameter to the C DLL.
The GrpcStubInterpreter should assign `read_array[:] = reponse.read_array`.

Reply:
Sure brad, i have created a task to track this [Task 2371633](https://ni.visualstudio.com/DevCentral/_workitems/edit/2371633): Handle the read functions with read array inputs in library interpreter